src/priceClient/adapters/across.ts

```diff
@@ -0,0 +1,60 @@
+import assert from "assert";
+import axios, { AxiosError } from "axios";
+import { msToS, PriceFeedAdapter, TokenPrice } from "../priceClient";
+
+type AcrossPrice = { price: number };
+const defaultTimeout = 5000; // mS
+
+export class PriceFeed implements PriceFeedAdapter {
+  protected _timeout = defaultTimeout;
+
+  constructor(public readonly name: string, public readonly host?: string) {
+    // Allow host to be overridden for test or alternative deployments.
+    if (this.host === undefined) this.host = "across.to";
+  }
+
+  get timeout(): number {
+    return this._timeout;
+  }
+
+  set timeout(timeout: number) {
+    assert(timeout >= 0);
+    this._timeout = timeout;
+  }
+
+  async getPriceByAddress(address: string, currency = "usd"): Promise<TokenPrice> {
+    // Note: Assume price is 60 seconds old since the API does not provide a timestamp.
+    const now = msToS(Date.now()) - 60;
+    const acrossPrice: AcrossPrice = await this.query(address, currency, this._timeout);
+
+    return { address: address, price: acrossPrice.price, timestamp: now };
+  }
+
+  // todo: Support bundled prices in the API endpoint.
+  async getPricesByAddress(addresses: string[], currency = "usd"): Promise<TokenPrice[]> {
+    const prices: TokenPrice[] = [];
+    for (const address of addresses) {
+      const price: TokenPrice = await this.getPriceByAddress(address, currency);
+      prices.push(price);
+    }
+
+    return prices;
+  }
+
+  private async query(token: string, baseCurrency = "usd", timeout: number = defaultTimeout): Promise<AcrossPrice> {
+    const url = `https://${this.host}/api/coingecko`;
+    const args = {
+      timeout,
+      params: {
+        l1Token: token,
+        baseCurrency: baseCurrency,
+      },
+    };
+
+    const result = await axios(url, args).catch((err) => {
+      const errMsg: string = err instanceof AxiosError ? err.message : "unknown error";
```

Comment:
nit: any reason why we can't just do err.toString() in the else case?

Reply:
Just my ignorance about the object that is provided during an exception. My impression is that it sometimes seems like an amorphous blob (!!), so I wasn't confident that I'd be able to interpret it reliably.

Reply:
Has been implemented in this commit btw: https://github.com/across-protocol/sdk-v2/pull/98/commits/fb5780a3c40637f17247612e7bbfc988319d3b9a